### ISSUE TITLE
CGCNNModel: more epochs != more accuracy

### ISSUE THREAD

**Issue by rkurchin** [deepchem/deepchem#2160]
I worked up something almost exactly analogous to [this](https://github.com/deepchem/deepchem/blob/28d0e19b2af07918bd12be42c12ea13ea44c5c4e/deepchem/models/torch_models/cgcnn.py#L245) just with importing/featurizing my own data from CIF files instead of doing the premade import. As far as I could verify, the featurization works fine, but I find that there's no improvement in accuracy when I train for more epochs. For example, in the last attempt, I saw MAE's of 0.88-0.89 across the board ranging from 5-50 epochs of training.

@nd-02110114, @rbharath mentioned you already noticed something similar and are working on it now; he asked me to file an issue here to coordinate. 

The code I'm running is below, but there are 30k+ CIF files it relies on so it might be hard to reproduce independently without sharing those somehow... (though they're all on Materials Project and I can share the code to download them yourself if you like, or, I imagine you'd see the same thing with whatever CIFs you use...)

```
import deepchem as dc
import pymatgen.core.structure as pmgs
from glob import glob
import pandas as pd
import numpy as np
from deepchem.splits.splitters import RandomSplitter

# construct an iterable over tuples (X, y, w, ids)
# X = output of featurize
# y = formation energy
# w = 1.0
# id = 'mp-#'

# TODO: this should yield a tuple of lists
def get_data(cif_dir, csv_path, chunk_size=8192):
  featurizer = dc.feat.CGCNNFeaturizer()
  cifs = glob(cif_dir+"*.cif")
  # shuffle list
  label_df = pd.read_csv(csv_path)
  total_size = len(label_df)
  num_chunks = int(np.ceil(total_size/chunk_size))
  cif_ind = 0
  for chunk in range(num_chunks):
    i = 0
    Xs = []
    ys = []
    ws = []
    ids = []
    while i < chunk_size:
      if cif_ind >= len(cifs):
        i = chunk_size
        continue
      cif = cifs[cif_ind]
      cif_ind += 1
      struc = pmgs.Structure.from_file(cif)
      X = featurizer.featurize([struc])[0]
      id = cif.split("/")[-1][:-4]
      label = label_df[label_df.task_id==id].formation_energy_per_atom
      if len(label)==0:
        continue
      else:
        i += 1
        Xs.append(X)
        ys.append(float(label))
        ws.append(1.0)
        ids.append(id)
    print("finished a chunk!")
    yield (np.array(Xs), np.array(ys), np.array(ws), np.array(ids))

# create dataset (these paths are on my local machine obviously...)
cif_dir = "MP_data/formation_energy_per_atom_cifs/"
csv_path = "MP_data/formation_energy_per_atom.csv"
it = get_data(cif_dir, csv_path)
#dataset = dc.data.DiskDataset.create_dataset(it)

# next: split into train, test, validate
s = RandomSplitter()
train, valid, test = s.train_valid_test_split(dataset)

# build model, just use all the defaults for now
model1 = dc.models.CGCNNModel(mode='regression', learning_rate=0.001)
model1.fit(train, nb_epoch=5) # or 10, 20, 50...
```

**Comment by rbharath** [comment]
This might be possibly linked to @nd-02110114's work on https://github.com/deepchem/deepchem/pull/2138. 

@nd-02110114 I recall you mentioned that you were also seeing some training issues with CGCNNs. Did you manage to figure out a workaround?

**Comment by nd-02110114** [comment]
> @nd-02110114 I recall you mentioned that you were also seeing some training issues with CGCNNs. Did you manage to figure out a workaround?

Yes. The currently CGCNNModel and GATModel has some issues for training. The loss value doesn't decrease well. So, I'm debugging now and I haven't understood what is the reason. I will focus on fixing this issue in the next week (9/13-9/20)

**Comment by nd-02110114** [comment]
I still debug for CGCNNModel. The problem is that the loss doesn't decrease after 1 epoch like this.

```
1.2885267893473307
0.5434092839558919
0.5397125244140625
0.5522617340087891
0.5471174240112304
0.5393458684285481
0.540155553817749
0.5367419560750325
0.5460977554321289
0.566059430440267
```

The notebook https://colab.research.google.com/drive/1bEsghp49n_GoQdZ44omYz2b9IQnO2-IB?usp=sharing

In this week, I'm trying to fix this problem. But, if i couldn't fix this issue, I may try the CGCNNModel with PyTorch Geometric. I found the good OSS implementation using PyTorch Geometric.
https://github.com/Open-Catalyst-Project/baselines/blob/5f32a65a2e95b19daf0ce37f7f4b6e5c1188fb62/ocpmodels/models/cgcnn.py

**Comment by nd-02110114** [comment]
I'm still debugging CGCNN model. Maybe, I think the `CGCNNFeaturizer` have some problems, but I couldn't understand clearly.

Please see the notebook. I wrote the details about my debug.
https://colab.research.google.com/drive/1bEsghp49n_GoQdZ44omYz2b9IQnO2-IB?usp=sharing
